go/libraries/doltcore/env/environment.go

```diff
@@ -650,3 +653,114 @@ func (dEnv *DoltEnv) GetUserHomeDir() (string, error) {
 func (dEnv *DoltEnv) TempTableFilesDir() string {
 	return filepath.Join(dEnv.GetDoltDir(), tempTablesDir)
 }
+
+// PutDocsToWorking adds, updates or removes dolt_docs table on the working root. The table will be added or updated
+// When at least one doc.NewerText != nil. If the `dolt_docs` table exists and every doc.NewerText == nil, the table will be removed.
+func (dEnv *DoltEnv) PutDocsToWorking(ctx context.Context, docDetails []*doltdb.DocDetails) error {
+	root, err := dEnv.WorkingRoot(ctx)
+
+	if err != nil {
+		return err
+	}
+
+	docTbl, found, err := root.GetTable(ctx, doltdb.DocTableName)
+
+	if err != nil {
+		return err
+	}
+
+	if found {
+		m, err := docTbl.GetRowData(ctx)
+		if err != nil {
+			return err
+		}
+
+		sch, err := docTbl.GetSchema(context.Background())
+		if err != nil {
+			return err
+		}
+
+		me := m.Edit()
+		for _, doc := range docDetails {
+			docRow, exists, err := docTbl.GetRowByPKVals(context.Background(), row.TaggedValues{doltdb.DocNameTag: types.String(doc.DocPk)}, sch)
+			if err != nil {
+				return err
+			}
+
+			if exists && *doc.NewerText == nil {
+				me = me.Remove(docRow.NomsMapKey(sch))
+			} else if *doc.NewerText != nil {
+				docTaggedVals := row.TaggedValues{
+					doltdb.DocNameTag: types.String(doc.DocPk),
+					doltdb.DocTextTag: types.String(*doc.NewerText),
+				}
+				docRow, err = row.New(types.Format_7_18, sch, docTaggedVals)
+				if err != nil {
+					return err
+				}
+				me = me.Set(docRow.NomsMapKey(sch), docRow.NomsMapValue(sch))
+			}
+		}
+		updatedMap, err := me.Map(ctx)
+		var newRoot *doltdb.RootValue
+		if updatedMap.Len() == 0 {
+			newRoot, err = root.RemoveTables(ctx, doltdb.DocTableName)
+			if err != nil {
+				return err
+			}
+		} else {
+			docTbl, err = docTbl.UpdateRows(ctx, updatedMap)
+			if err != nil {
+				return err
+			}
+			newRoot, err = root.PutTable(ctx, doltdb.DocTableName, docTbl)
+			if err != nil {
+				return err
+			}
+		}
+		return dEnv.UpdateWorkingRoot(ctx, newRoot)
+	}
+
+	// Create a dolt_docs table if it doesn't exist yet
```

Comment:
move to a function called createDocsTable

Reply:
done!